test/tools/router_check/router.cc

```diff
@@ -16,6 +16,46 @@
 
 #include "test/test_common/printers.h"
 
+namespace {
+const std::string
+toString(envoy::config::route::v3::HeaderMatcher::HeaderMatchSpecifierCase specifier) {
+  switch (specifier) {
+  case envoy::config::route::v3::HeaderMatcher::HeaderMatchSpecifierCase::kExactMatch:
+    return "exact_match";
+    break;
+  case envoy::config::route::v3::HeaderMatcher::HeaderMatchSpecifierCase::
+      kHiddenEnvoyDeprecatedRegexMatch:
+    return "regex_match";
+    break;
+  case envoy::config::route::v3::HeaderMatcher::HeaderMatchSpecifierCase::kSafeRegexMatch:
+    return "safe_regex_match";
+    break;
+  case envoy::config::route::v3::HeaderMatcher::HeaderMatchSpecifierCase::kRangeMatch:
+    return "range_match";
+    break;
+  case envoy::config::route::v3::HeaderMatcher::HeaderMatchSpecifierCase::kPresentMatch:
+  case envoy::config::route::v3::HeaderMatcher::HeaderMatchSpecifierCase::
+      HEADER_MATCH_SPECIFIER_NOT_SET:
+    return "present_match";
+    break;
+  case envoy::config::route::v3::HeaderMatcher::HeaderMatchSpecifierCase::kPrefixMatch:
+    return "prefix_match";
+    break;
+  case envoy::config::route::v3::HeaderMatcher::HeaderMatchSpecifierCase::kSuffixMatch:
+    return "suffix_match";
+    break;
+  default:
+    return "HEADER_MATCH_SPECIFIER_NOT_SET";
+    break;
+  }
```

Comment:
Can you remove the default? This will make sure this gets updated if a new one is added. I think you have them all covered with the NOT_SET above.